### PR TITLE
Add protoc gen doc support

### PR DESCRIPTION
**Pull request by kissmikijr**
## Hey, I just made a Pull Request!
This is a continuation fo this PR: https://github.com/backstage/backstage/pull/11746

Adds the possibility to render a generated [protoc-gen-doc](https://github.com/pseudomuto/protoc-gen-doc) descriptors inside the gRPC API widget.

This uses the [grpc-docs](https://github.com/gendocu-com/grpc-docs) package to render the generated files.

With an entity setup:
```yaml
apiVersion: backstage.io/v1alpha1
kind: API
metadata:
  name: hello-world
  description: Hello World example for gRPC
spec:
  type: grpc-docs
  lifecycle: deprecated
  owner: team-c
  definition:
    $text: https://github.com/myOrg/myRepo/my-descriptor.json

```
An example gendocu json rendered in the apiwidget:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by marcus-crane** [comment]
Ah, very cool!

File: plugins/api-docs-react/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@backstage/plugin-api-docs-react",
```
[the remark this answers is not in the document]
Comment (kissmikijr) [reply plugins/api-docs-react/package.json:2]: Ah, yea make sense! I wasn't sure so I just went with a react package! Thank you for the explanation!

**Review by Rugvip — APPROVED** [review]
Nice, let's :shipit:!

**Comment by marcus-crane** [comment]
🤩

**Pull request merged**